File: Python/RecursivePower.py
def power(x, n):
  if n == 0:
    return 1
  elif n < 0:
    return 1/power(x, -n)
  elif n%2 != 0:
    return x * power(x,n-1)
  elif n%2 == 0:
    z = power (x, n/2)
    return z*z


def displayPower(x, n):
    print(f"{x} to the power of {n} is {power(x, n)}")

# test
displayPower(3, 2)

File: JS/RecursivePower.js
var isEven = function(n) {return n % 2 === 0;};
var isOdd = function(n) {return !isEven(n);};
var power = function(x, n) {
    println("Computing " + x + " raised to power " + n + ".");
    if (n === 0){return 1;} // base case
    else if (n < 0){return 1/power(x,-n);} // recursive case: n is negative 
    else if (isOdd(n)){return x*(power(x,n-1));} // recursive case: n is odd
    else if (isEven(n)){
        var z = power(x,n/2);
        return z*z
        ;} // recursive case: n is even
};

var displayPower = function(x, n) {
    println(x + " to the " + n + " is " + power(x, n));
};

displayPower(3, 0);
Program.assertEqual(power(3, 0), 1);
